tests/baselines/reference/assignmentToObjectAndFunction.errors.txt

```diff
@@ -14,6 +16,7 @@ tests/cases/compiler/assignmentToObjectAndFunction.ts(29,5): error TS2322: Type
 !!! error TS2322: Type '{ toString: number; }' is not assignable to type 'Object'.
 !!! error TS2322:   Types of property 'toString' are incompatible.
 !!! error TS2322:     Type 'number' is not assignable to type '() => string'.
+!!! error TS2322:       Type 'Number' provides no match for the signature '(): string'

```

Comment:
I think you might want to check against `TypeFlags.Primitive`. See how that works out.

In general, if type aliases aren't picked up that's not a blocker, but it'd be great if we knew that there's a test to check for the behavior.